main.go

```diff
@@ -132,23 +42,18 @@ func main() {
 		logger.Fatal(inputErr)
 	}
 
-	content, readErr := os.ReadFile(args.Plan)
-	if readErr != nil {
-		logger.Fatal("invalid plan JSON file")
-	}
-
-	var plan Plan
-	if unmarshErr := json.Unmarshal(content, &plan); unmarshErr != nil {
-		logger.Fatal("invalid plan JSON file")
+	plan, err := terraform.NewPlan(args.Plan)
+	if err != nil {
+		logger.Fatal(err)
 	}
 
 	result := Result{Ok: true, Errors: []ResultError{}}
 
-	requester := findExternalIdentity(args.Requester, &plan)
-	approvers := findApprovers(args.Approvers, args.Requester, &plan)
+	requester := findExternalIdentity(args.Requester, plan)
```

Comment:
This was probably the only "change" in the PR. Why did the original version work? Plan was already a pointer so the original implementation was a pointer to pointer. Why did it work / why was this change redundant?

Reply:
NewPlan returns a pointer. Adding & would mean we'd pass a pointer to a pointer (**Plan)

Reply:
Ah, yes ok now I get how it was changed.